modules/metastorage/src/testFixtures/java/org/apache/ignite/internal/metastorage/server/SimpleInMemoryKeyValueStorage.java

```diff
@@ -568,12 +585,80 @@ public void close() {
 
     @Override
     public CompletableFuture<Void> snapshot(Path snapshotPath) {
-        throw new UnsupportedOperationException();
+        synchronized (mux) {
+            try {
+                Files.createDirectories(snapshotPath);
+
+                Path snapshotFile = snapshotPath.resolve(SimpleInMemoryKeyValueStorageSnapshot.FILE_NAME);
+
+                assertTrue(IgniteUtils.deleteIfExists(snapshotFile), snapshotFile.toString());
+
+                Files.createFile(snapshotFile);
+
+                var revsIdxCopy = new HashMap<Long, Map<byte[], ValueSnapshot>>();
+                revsIdx.forEach((revision, entries) -> {
+                    Map<byte[], ValueSnapshot> entries0 = new HashMap<>();
+                    entries.forEach((keyBytes, value) -> entries0.put(keyBytes, new ValueSnapshot(value)));
+
+                    revsIdxCopy.put(revision, entries0);
+                });
+
+                var snapshot = new SimpleInMemoryKeyValueStorageSnapshot(
+                        Map.copyOf(keysIdx),
+                        Map.copyOf(tsToRevMap),
+                        Map.copyOf(revToTsMap),
+                        revsIdxCopy,
+                        rev,
+                        updCntr,
+                        savedCompactionRevision
+                );
+
+                byte[] snapshotBytes = ByteUtils.toBytes(snapshot);
+
+                Files.write(snapshotFile, snapshotBytes, WRITE);
+
+                return nullCompletedFuture();
```

Comment:
I would suggest dispatching this process onto an executor in order to more closely mimic the production behaviour

Reply:
I think I'll leave it as is.

Reply:
Because this class will be used in tests and having a truly async method will more closely emulate the behaviour of the production version

Reply:
But why ?

Reply:
This is a test implementation that must be correct in terms of data, but does not have to be a complete copy in behavior.
At the beginning you suggested, your arguments did not convince me, I will leave it as is.

Reply:
> but does not have to be a complete copy in behavior.

But this is literally the contract of the interface, this method is expected to be async, what's so difficult in adding `runAsync` around this method?

Reply:
I have two arguments here:

1. The contract of this method expects it to be async;
2. This will more closely mimic the production behaviour;
3. It's just a couple of lines of code

Your only argument is "I don't want to". Can we be more constructive here?

Reply:
The product implementation uses an external executor, the test one does not. And it is not clear what kind of executor should be created and passed to all tests, it looks like a change of more than one line of code and a file for such a task. Using a common pool seems dangerous to me, because we can hang one thread and other tests may suffer because of it. Since they will not have enough threads in the common pool. You can say that this is unlikely to happen, but you cannot refute that this cannot but happen.